drivers/gles3/shader_tracker_gles3.cpp

```diff
@@ -0,0 +1,445 @@
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "shader_tracker_gles3.h"
+#include "core/crypto/crypto_core.h"
+#include "core/io/compression.h"
+#include "core/io/file_access_compressed.h"
+#include "core/os/dir_access.h"
+#include "core/os/file_access.h"
+#include "core/os/os.h"
+#include "core/ustring.h"
+#include "rasterizer_storage_gles3.h"
+
+#include "servers/visual/visual_server_globals.h"
+
+const String SPATIAL_LIST_FILE_NAME = "spatial.shaderlist";
+const String CANVAS_LIST_FILE_NAME = "canvas.shaderlist";
+const String PARTICLE_LIST_FILE_NAME = "particle.shaderlist";
+
+const String SHADER_CODE_MARKER = "SHADER_CODE:";
+const String LIST_CODE_SEPERATOR = "SHADER_LIST_START";
+
+ShaderTrackerGLES3::ShaderTrackerGLES3() {
+	spatial_shader_code = Map<uint32_t, String>();
+	canvas_shader_code = Map<uint32_t, String>();
+	particle_shader_code = Map<uint32_t, String>();
+
+	used_spatial_shaders = Map<uint32_t, Map<uint32_t, String>>();
+	used_canvas_shaders = Map<uint32_t, Map<uint32_t, String>>();
+	used_particle_shaders = Map<uint32_t, Map<uint32_t, String>>();
+
+	output_folder = OS::get_singleton()->get_executable_path().get_base_dir();
+}
+
+ShaderTrackerGLES3::~ShaderTrackerGLES3() {
+	FileAccessCompressed *fa_spatial = memnew(FileAccessCompressed);
```

Comment:
In cases like this you can just declare a stack-lived variable: `FileAccessCompressed fa_spatial;`.

You can handle its lifetime either by scoping or just reusing the same variable.